### PR TITLE
Install and configure lmm for database snapshots

### PR DESCRIPTION
**Pull request by deviantintegral**
Following up from https://github.com/laravel/homestead/issues/1071

This PR moves the MySQL database directory to a thinly-provisioned LVM volume so we can support copy-on-write database snapshots.

Open questions:

* Is 10GB the right size? I've only ever had one project go over that, but in the current homestead you could import a 30GB database without issue. It's intentionally smaller than the 40GB thin volume so you could have up to 4 completely distinct snapshots at 10GB each. We could also increase the base VM disk size. Other than the free space reduction, there's no affect for users who don't need snapshots.
* How do we want to handle the `guided_size` parameter? We could copy the preseed files into our repository, and copy or hard link them. However, with multiple files to manage I wonder if it's better to fork and merge all of bento into this repository, pulling from upstream as needed.

```diff
diff --git a/ubuntu/http/preseed-hyperv.cfg b/ubuntu/http/preseed-hyperv.cfg
index 7f554b8..9670a79 100755
--- a/ubuntu/http/preseed-hyperv.cfg
+++ b/ubuntu/http/preseed-hyperv.cfg
@@ -9,7 +9,7 @@ d-i mirror/country string manual
 d-i mirror/http/directory string /ubuntu/
 d-i mirror/http/hostname string archive.ubuntu.com
 d-i mirror/http/proxy string
-d-i partman-auto-lvm/guided_size string max
+d-i partman-auto-lvm/guided_size string 20G
 d-i partman-auto/choose_recipe select atomic
 d-i partman-auto/method string lvm
 d-i partman-lvm/confirm boolean true
diff --git a/ubuntu/http/preseed.cfg b/ubuntu/http/preseed.cfg
index b13ca74..8026f84 100644
--- a/ubuntu/http/preseed.cfg
+++ b/ubuntu/http/preseed.cfg
@@ -9,7 +9,7 @@ d-i mirror/country string manual
 d-i mirror/http/directory string /ubuntu/
 d-i mirror/http/hostname string archive.ubuntu.com
 d-i mirror/http/proxy string
-d-i partman-auto-lvm/guided_size string max
+d-i partman-auto-lvm/guided_size string 20GB
 d-i partman-auto/choose_recipe select atomic
 d-i partman-auto/method string lvm
 d-i partman-lvm/confirm boolean true
```

### CODE REVIEW

File: scripts/provision.sh [hunk 1]
```diff
@@ -271,6 +271,36 @@ echo "mysql-server mysql-server/root_password password secret" | debconf-set-sel
 echo "mysql-server mysql-server/root_password_again password secret" | debconf-set-selections
 apt-get install -y mysql-server
 
+# Install LMM for database snapshots
+apt-get install -y thin-provisioning-tools bc
+git clone -b ubuntu-18.04 https://github.com/Lullabot/lmm.git /opt/lmm
```
Comment (deviantintegral) [inline scripts/provision.sh]: This branch upstream:

* Replaces `service` with `systemctl` to support Ubuntu 18.04 (and I guess 16.04 as well)
* Removes the requirement for a dedicated volume group. Packer doesn't support multiple disks in a provider-independent way, so instead there is a hardcoded `mysql-` prefix for all mysql volumes. There's very little that is actually tied to mysql or mariadb, so if we wanted to support postgres or mongo we could.

**Comment by svpernova09** [comment]
Is this ready for me to poke around or are you still working on changes?

**Comment by deviantintegral** [comment]
It should be ready for review. There's still one or two more followup commits to do based on my comments in the PR description, so let me know what you think.

**Comment by svpernova09** [comment]
>    Is 10GB the right size? I've only ever had one project go over that, but in the current homestead you could import a 30GB database without issue. It's intentionally smaller than the 40GB thin volume so you could have up to 4 completely distinct snapshots at 10GB each. We could also increase the base VM disk size. Other than the free space reduction, there's no affect for users who don't need snapshots.

Let's start at 10GB and go from there. The base box *should* dynamically grow until it hits 120GB I believe is the hard limit but it's been a while since I've tested/verified that

>    How do we want to handle the guided_size parameter? We could copy the preseed files into our repository, and copy or hard link them. However, with multiple files to manage I wonder if it's better to fork and merge all of bento into this repository, pulling from upstream as needed.

Yeeeeeeah....

It's gross, but lets use `sed` in the `link-to-bento.sh` script to link our homestead.sh but also make the change of `cleanup.sh` -> `homestead.sh` in addition to the `guided_size` to the value we want.

**Comment by svpernova09** [comment]
I think this answers all the outstanding questions. if not keep bugging me!

**Comment by deviantintegral** [comment]
Done! I handled using `gsed` for macOS and BSD (I assume anyone building this will have that available).

**Comment by svpernova09** [comment]
Here's what I'm getting:

```
sh bin/link-to-bento.sh                                                                                                                                                           
bin/link-to-bento.sh: line 7: gsed: command not found
```

**Comment by deviantintegral** [comment]
I was missing quotes around the string to test against `-x`. I've added them, and also made the whole block specific to macOS.

**Comment by svpernova09** [comment]
> On macOS this script requires GNU sed. Install it with 'brew install gnu-sed'.

Not trying to be difficult here but can we use the built in sed? Granted the people who will *ever* run this command is less than 10 people, I'd rather not have to brew install something to do what we're doing.

**Comment by deviantintegral** [comment]
No, good call, and on a closer read of the man page I realized how to make it compatible with both flavours of sed. Many lines deleted!

I went to update from master and the conflicts look to be from some refactoring work to use amazon linux as the base instead of ubuntu. Was that meant to go into master?

One tangent - I recently was on a project that was deployed with amazon linux, and we ran into some tricky issues with PHP. In particular, they were packaging php extensions like `igbinary` for PHP 7.2 from an ancient master commit that wasn't even compatible with PHP 7.1. It was pretty clear they were just checking compile + load without actually _using_ the extensions. We ended up switching to building all of PHP from source.

**Comment by svpernova09** [comment]
> I went to update from master and the conflicts look to be from some refactoring work to use amazon linux as the base instead of ubuntu. Was that meant to go into master?

No, that was a really terrible mistake on my end with another project I'm working on. I got my paths wrong and ended up pushing a change. That commit has been resolved and you shouldn't have any conflicts now.

**Comment by svpernova09** [comment]
Thanks!

**Pull request merged**